File: 0908-middle-of-the-linked-list/solution.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* middleNode(ListNode* head) {
        
        ListNode *n = head , *m = head ; 
        while(m != NULL){
         m = m->next ; 
            if( m == NULL)return n ; 
         n = n->next ;
         m = m-> next ;
        }
        return n ;
    }
};

File: 0908-middle-of-the-linked-list/solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {

    public ListNode middleNode(ListNode head) {
        ListNode slow = head ;
        ListNode fast = head ;

        while(fast  != null && fast.next != null){ // odd and even Right covered in this
            slow = slow.next ;
            fast = fast.next.next ;
        } // slow pointer is the answer 

        return slow ;
    }
}
